components/Discussion/enhancers.js

```diff
@@ -365,3 +366,79 @@ mutation discussionSubmitComment($discussionId: ID!, $parentId: ID, $id: ID!, $c
     }
   })
 })
+
+const discussionPreferencesQuery = gql`
+query discussionPreferences($discussionId: ID!) {
+  me {
+    id
+    publicUser {
+      credentials {
+        description
+      }
+    }
+  }
+  discussion(id: $discussionId) {
+    id
+    rules {
+      maxLength
+      minInterval
+      anonymity
+    }
+    userPreference {
+      anonymity
+      credential {
+        description
+        verified
+      }
+    }
+  }
+}
+`
+export const withDiscussionPreferences = graphql(discussionPreferencesQuery, {
+  variables: ({discussionId}) => ({discussionId})
```

Comment:
Just an apollo newbie question:
I thought `variables` only appear within `{options: variables: {...}} `of a graphql() call, or within the data prop: `{ data: { variables } }`. Does it actually do anything here, or does it just work because graphql reads the `discussionId` from the component's ownProps anyway?

Reply:
I'll try removing the code and see if it still works :)